aspnet/fundamentals/localization.rst

```diff
@@ -1,8 +1,302 @@
-.. include:: /../common/stub-topic.txt
+ASP.NET Core Internationalization, Localization and Globalization 
+==================================================================
 
-|stub-icon| Localization
-========================
+By `Rick Anderson`_ with assistance from `Damien Bowden <https://twitter.com/damien_bod>`_, `Bart Calixto @bartmax <https://twitter.com/bartmax>`_, and `Nadeem Afana <https://twitter.com/NadeemAfana>`__ 
 
-.. include:: /../common/stub-notice.txt
+Creating a multilingual website withe ASP.NET Core is not easy but it will allow your site to reach a wider audience. ASP.NET  has :doc:`/fundamentals/request-features` that provide support for localizing into different languages and cultures.
+
+This sample is based on the ``Localization.StarterWeb`` project on `GitHub <https://github.com/aspnet/entropy>`__.
+
+.. contents:: In this article:
+  :local:
+  :depth: 1
+  
+.. note:: Currently, resource files are not read when the project is run from Visual Studio. See `this issue <https://github.com/aspnet/dnx/issues/3047>`_ for more information.
+
+IStringLocalizer
+---------------------------------------------
+
+Introduced in ASP.NET Core, ``IStringLocalizer`` was architected to maximize productivity when developing localized apps. 
+
+The simple interface has an indexer and an ``IEnumerable`` for returning localized strings. Unlike previous localization approaches, ``IStringLocalizer`` doesn't require you to store the default language strings in a resource file. You can develop an app targeted for localization and not need to create resource files early in development. The recommendations is that you do **not** create a default language resource file. In previous version of ASP.NET, you had to move all your localization strings into a resource (.resx) file and change your app everywhere the strings were used, even for the default language. You no longer need to move your localized strings, you simply wrap them.
+
+.. literalinclude:: localization/sample/Controllers/AboutController.cs
+  :language: c#
+
+In the code above, the ``IStringLocalizer`` implementation comes from :doc:`/fundamentals/dependency-injection`. If the localized value of "About Title" is not found, then the indexer key is returned, that is, the string "About Title". The recommended work flow is to leave the default language literal strings in the app and wrap them in the localizer. You can develop your app with your default language and prepare it for the localization step without first creating a default resource file. You can use the traditional approach and provide a key to retrieve the default language string, but the new workflow is optimized for reducing the overhead of localizing an app.
+
+Use the ``IHtmlLocalizer`` implementation for resources that contain markup. ``IHtmlLocalizer`` encodes arguments that are formatted in the resource string, but not the resource string. In the sample highlighted below, only the ``name`` parameter is encoded.
```

Comment:
I think you are describing just one workflow, this shouldn't be the only nor the recommended one. Maybe note what are the pros of this one. Definitely have some cons too. Pro's should be that you can create your site without a localization strategy from the beginning or using keys as values.

some cons are:
long text are harder to work with
updating texts may became a burden if you use the key as of the default language.